src/geometry/mesh.rs

```diff
@@ -12,7 +12,7 @@ pub struct Mesh<V, S> {
     bounding_volume: Box<V>,
 }
 
-impl<'a, V: BoundingVolume, S: TriangleStorage<'a>> Mesh<V, S> {
+impl<'a, V: BoundingVolume, S: 'a + TriangleStorage<'a>> Mesh<V, S> {
```

Comment:
Why the `'a +` here?

Reply:
At some point the compiler suggested it fixed a compilation error 🤷‍♂️